homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,231 @@
+            _LOGGER.warning(
+                "Your flux_led configuration has already been imported. Please remove configuration from your configuration.yaml"
+            )
+            return self.async_abort(reason="single_instance_allowed")
+
+        _LOGGER.warning(
+            "Imported auto_add configuration for flux_led. Please remove from your configuration.yaml"
+        )
+        return await self.async_step_user(
+            user_input={
+                CONF_AUTOMATIC_ADD: data[CONF_AUTOMATIC_ADD],
+                CONF_DEVICES: data.get(CONF_DEVICES, {}),
+            }
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        config_entry = self.hass.config_entries.async_entries(DOMAIN)
+        if config_entry:
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            devices = user_input.get(CONF_DEVICES, {})
+
+            if user_input[CONF_AUTOMATIC_ADD]:
+                scanner = BulbScanner()
+                await self.hass.async_add_executor_job(scanner.scan)
+
+                for bulb in scanner.getBulbInfo():
+                    device_id = bulb["ipaddr"].replace(".", "_")
+                    if not devices.get(device_id, False):
+                        devices[device_id] = {
+                            CONF_NAME: bulb["ipaddr"],
+                            CONF_HOST: bulb["ipaddr"],
+                        }
+
+            return self.async_create_entry(
+                title="FluxLED/MagicHome",
+                data={
+                    CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+                    CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
+                    CONF_DEVICES: devices,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_AUTOMATIC_ADD, default=True): bool}
+            ),
+            errors=errors,
+        )
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle flux_led options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize the flux_led options flow."""
+
+        self._config_entry = config_entry
+        self._global_options = None
+        self._device_registry = None
+        self._configure_device = None
+        self._title = "FluxLED/MagicHome"
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        return await self.async_step_prompt_options()
+
+    async def async_step_prompt_options(self, user_input=None):
+        """Manage the options."""
+
+        errors = {}
+
+        if user_input is not None:
+            self._global_options = {
+                CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+                CONF_EFFECT_SPEED: user_input[CONF_EFFECT_SPEED],
```

Comment:
Where is the global speed option used?

Reply:
My mistake - forgot to include it. Light effect speed is set now based on 1) device specific speed if set, 2) global options speed if set, 3) default speed.